bookkeeper-server/src/test/java/org/apache/bookkeeper/client/MockLedgerHandle.java

```diff
@@ -100,7 +108,7 @@ public void run() {
                 final Queue<LedgerEntry> seq = new ArrayDeque<LedgerEntry>();
                 long entryId = firstEntry;
                 while (entryId <= lastEntry && entryId < entries.size()) {
-                    seq.add(entries.get((int) entryId++));
+                    seq.add(new LedgerEntry(entries.get((int) entryId++).duplicate()));
```

Comment:
there is a memory leak here, you don't need duplicate.

old `LedgerEntry` constructor will retain the buffer.